sky/clouds/kubernetes.py

```diff
@@ -109,17 +109,19 @@ def _unsupported_features_for_resources(
     def max_cluster_name_length(cls) -> Optional[int]:
         return cls._MAX_CLUSTER_NAME_LEN_LIMIT
 
-    @classmethod
-    def regions(cls) -> List[clouds.Region]:
-        return cls._regions
-
     @classmethod
     def regions_with_offering(cls, instance_type: Optional[str],
                               accelerators: Optional[Dict[str, int]],
                               use_spot: bool, region: Optional[str],
                               zone: Optional[str]) -> List[clouds.Region]:
-        # No notion of regions in Kubernetes - return a single region.
-        return cls.regions()
+        allowed_contexts = skypilot_config.get_nested(
+            ('kubernetes', 'allowed_contexts'), None)
+        if allowed_contexts is None:
+            return cls._regions
```

Comment:
Conceptually, I found it more clear to have the following mapping:
k8s contexts -> local cloud config profiles. 
Because both of them contains:
1. the identity to use for accessing the resource pool (k8s: user + namespace; cloud config: account)
2. the resource pool to look at (k8s: cluster; cloud config: project to use)

I think the current way is a simple workaround for now, but we may need to have a better design. The main confusion with using region may come from: multiple context can map to the same k8s cluster with different namespace or user.

Reply:
Agreed, probably need better solutions. I just realized many properties in the config may need to be updated in the near future to work well for multi-cluster (e.g., some contexts may need `ports: ingress`, while others may need `ports: loadbalancer`. Same ofr other fields).

Reply:
Yes, when I am updating the code for always showing context for region, I also realized that there are more places to be updated, especially the code for failover `Kubernetes._get_feasible_launchable_resources`. If we have two clusters with different resource set, our failover will likely disregard all the Kubernetes clusters if the cluster without the resource is the current activate context.

Marking this PR to draft for now to fix this issue.